app/src/main/java/com/example/android/uamp/fragments/NowPlayingScreen.kt

```diff
@@ -0,0 +1,220 @@
+ * @param nowPlayingFragmentViewModel to reference NowPlayingFragmentViewModel functions
+ * @param mainActivityViewModel to reference MainActivityViewModel functions
+ * @param navController to navigate to SettingsScreen
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun NowPlayingDescription(
+    mediaItem: MediaItem,
+    nowPlayingFragmentViewModel: NowPlayingFragmentViewModel,
+    mainActivityViewModel: MainActivityViewModel,
+    navController: NavController
+) {
+    if (mediaItem.equals(null)) {
+        Text(text = "Media Item is null")
+    } else {
+        Surface() {
+            NowPlaying(mediaItem, nowPlayingFragmentViewModel, mainActivityViewModel, navController)
+        }
+    }
+}
+
+/**
+ * NowPlaying describes the UI of the NowPlayingScreen which recomposes upon changes in the state of
+ * currentMediaItem's position, duration, and change in playback (play/pause).
+ *
+ * @param mediaItem current mediaItem being played
+ * @param nowPlayingFragmentViewModel to reference NowPlayingFragmentViewModel functions
+ * @param mainActivityViewModel to reference MainActivityViewModel functions
+ * @param navController to navigate to SettingsScreen
+ */
+@SuppressLint("PrivateResource")
+@Composable
+private fun NowPlaying(
+    mediaItem: MediaItem,
+    nowPlayingFragmentViewModel: NowPlayingFragmentViewModel,
+    mainActivityViewModel: MainActivityViewModel,
+    navController: NavController
+) {
+    val position: Long by nowPlayingFragmentViewModel.mediaPosition.observeAsState(0)
+    val duration: Long by nowPlayingFragmentViewModel.mediaDuration.observeAsState(0)
+    val buttonRes: Int? by nowPlayingFragmentViewModel.mediaButtonRes.observeAsState()
+
+    val buttonWidth = dimensionResource(id = R.dimen.exo_media_button_width)
+    val margins = dimensionResource(id = R.dimen.text_margin)
+
+    val mediaItemArtwork = mediaItem.mediaMetadata.artworkUri
+
+    Column(modifier = Modifier.fillMaxSize()) {
+        TopAppBar(
+            title = { Text("") },
+            actions = {
+                IconButton(onClick = {
+                    navController.navigate("settings")
+                }) {
+                    Icon(Icons.Filled.Settings, contentDescription = null)
+                }
+            },
+            backgroundColor = colorResource(id = R.color.colorPrimary)
+        )
+
+        AsyncImage(
+            model = mediaItemArtwork,
+            contentDescription = stringResource(id = R.string.album_art_alt),
+            modifier = Modifier
+                .weight(2f)
+                .fillMaxWidth()
+        )
+
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .background(colorResource(id = R.color.colorPrimary)),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Image(
+                painter = painterResource(id = buttonRes!!),
+                contentDescription = null,
```

Comment:
This should definitely have a content description and it should be based on the current state (even more reason not to have the view model passing the icon). A visually impaired user needs to know if the button will play or pause too :)